lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -108,6 +132,35 @@ static bool canDeriveConformance(DeclContext *DC,
   return false;
 }
 
+/// Diagnose failed conformance synthesis caused by a member type not conforming
+/// to the same protocol
+void diagnoseFailedDerivation(DeclContext *DC, NominalTypeDecl *nominal,
+                              ProtocolDecl *protocol) {
+  ASTContext &ctx = DC->getASTContext();
+
+  if (auto *enumDecl = dyn_cast<EnumDecl>(nominal)) {
+    auto nonConformingAssociatedTypes =
+        associatedValuesNotConformingToProtocol(DC, enumDecl, protocol);
+    for (auto *typeToDiagnose : nonConformingAssociatedTypes) {
+      ctx.Diags.diagnose(
+          typeToDiagnose->getTypeLoc().getLoc(),
+          diag::missing_member_type_conformance_prevents_synthesis, false,
```

Comment:
Define an enum in place of using a boolean here for readability

Reply:
👍 , fixed